src/+mpecopt/Solver.m

```diff
@@ -89,8 +93,11 @@
             % ------------------ Prepare homotopy solver for Phase I -----------------------------
             if strcmp(opts.initialization_strategy,"RelaxAndProject")
                 t_generate_nlp_solvers = tic;
-                % TODO : input just mpec_casadi and solver_initialization
-                [solver_relaxed,x_k_init,p0_relaxed,lbx_relaxed,ubx_relaxed,lbg_relaxed,ubg_relaxed] = create_phase_i_nlp_solver(mpec_casadi.f,mpec_casadi.g,mpec_casadi.x,mpec_casadi.x1,mpec_casadi.x2,mpec_casadi.p,solver_initialization.lbx,solver_initialization.ubx,solver_initialization.lbg,solver_initialization.ubg,solver_initialization.p0,x_k,opts,dims);
+                % !!!! TODO : create_phase_i_nlp_solver_dev - SHOULD be a method and executed outside of the solution loops (PROBABLY it can be executed while creating mpec_casadi)
+                % (which should also be a class itself)
+                [solver_relaxed, solver_initialization_relaxed] = create_phase_i_nlp_solver_dev(mpec_casadi,solver_initialization,opts,dims);
```

Comment:
even though here the correcty `solver_initialization` is passed, `solver_relaxed` produces an unbounded solution in the cardinality example. some other simpler examples work. 
I could not find what did introduce this bug in this PR.

Reply:
note also the todo in the comment in lines 96-97, if done quickly, would be appreciated.

Reply:
This bug is fixed and the solver init move is done.